Spigot-Server-Patches/0315-EntityTransformedEvent.patch

```diff
@@ -0,0 +1,103 @@
+From 104c99464e2091dd6dcf6d574ffa3656bc62cf70 Mon Sep 17 00:00:00 2001
+From: Anthony MacAllister <anthonymmacallister@gmail.com>
+Date: Thu, 26 Jul 2018 15:30:03 -0400
+Subject: [PATCH] EntityTransformedEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityMushroomCow.java b/src/main/java/net/minecraft/server/EntityMushroomCow.java
+index c6babe6..928a2b9 100644
+--- a/src/main/java/net/minecraft/server/EntityMushroomCow.java
++++ b/src/main/java/net/minecraft/server/EntityMushroomCow.java
+@@ -24,6 +24,16 @@ public class EntityMushroomCow extends EntityCow {
+ 
+             return true;
+         } else if (itemstack.getItem() == Items.SHEARS && this.getAge() >= 0) {
++            // Paper start
++            EntityCow entitycow = new EntityCow(this.world);
++            entitycow.setPositionRotation(this.locX, this.locY, this.locZ, this.yaw, this.pitch);
++            entitycow.setHealth(this.getHealth());
++            entitycow.aQ = this.aQ;
++            if (this.hasCustomName()) {
++                entitycow.setCustomName(this.getCustomName());
++            }
++            if (!new com.destroystokyo.paper.event.entity.EntityTransformedEvent(this.getBukkitEntity(), entitycow.getBukkitEntity(), com.destroystokyo.paper.event.entity.EntityTransformedEvent.TransformedReason.SHEARED).callEvent()) return false;
++            // Paper end
+             // CraftBukkit start
+             PlayerShearEntityEvent event = new PlayerShearEntityEvent((org.bukkit.entity.Player) entityhuman.getBukkitEntity(), this.getBukkitEntity());
+             this.world.getServer().getPluginManager().callEvent(event);
+@@ -35,15 +45,16 @@ public class EntityMushroomCow extends EntityCow {
+             this.die();
+             this.world.addParticle(Particles.u, this.locX, this.locY + (double) (this.length / 2.0F), this.locZ, 0.0D, 0.0D, 0.0D);
+             if (!this.world.isClientSide) {
+-                EntityCow entitycow = new EntityCow(this.world);
+-
+-                entitycow.setPositionRotation(this.locX, this.locY, this.locZ, this.yaw, this.pitch);
+-                entitycow.setHealth(this.getHealth());
+-                entitycow.aQ = this.aQ;
+-                if (this.hasCustomName()) {
+-                    entitycow.setCustomName(this.getCustomName());
+-                }
+-
++                // Paper start - moved up
++                //EntityCow entitycow = new EntityCow(this.world);
```

Comment:
Use a multiline comment (/* */) can reduce patch size.

Reply:
Did that once and was told by core team member(s) to explicitly comment each line when doing this.

Reply:
It really depends on whether we want to force a conflict should this area change.

In this case, because we're moving the logic around, we probably *do* want to force a conflict so that we can catch any future changes in this area.